infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/ImagePullSecretProvisioner.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.provision;
+
+import com.google.common.collect.ImmutableList;
+import com.google.gson.Gson;
+import com.google.gson.stream.JsonWriter;
+import io.fabric8.kubernetes.api.model.LocalObjectReference;
+import io.fabric8.kubernetes.api.model.LocalObjectReferenceBuilder;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.SecretBuilder;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.Base64;
+import java.util.List;
+import javax.inject.Inject;
+import org.eclipse.che.api.core.model.workspace.runtime.RuntimeIdentity;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.infrastructure.docker.client.UserSpecificDockerRegistryCredentialsProvider;
+import org.eclipse.che.infrastructure.docker.client.dto.AuthConfigs;
+import org.eclipse.che.workspace.infrastructure.kubernetes.KubernetesClientFactory;
+import org.eclipse.che.workspace.infrastructure.kubernetes.environment.KubernetesEnvironment;
+import org.eclipse.che.workspace.infrastructure.kubernetes.namespace.KubernetesNamespaceFactory;
+
+/** @author David Festal */
+public class ImagePullSecretProvisioner implements ConfigurationProvisioner<KubernetesEnvironment> {
+
+  static final String SECRET_NAME = "workspace-private-registries";
+
+  private final UserSpecificDockerRegistryCredentialsProvider
+      userSpecificDockerRegistryCredentialsProvider;
+  private final KubernetesClientFactory clientFactory;
+  private final KubernetesNamespaceFactory namespaceFactory;
+
+  @Inject
+  public ImagePullSecretProvisioner(
+      UserSpecificDockerRegistryCredentialsProvider userSpecificDockerRegistryCredentialsProvider,
+      KubernetesClientFactory clientFactory,
+      KubernetesNamespaceFactory namespaceFactory) {
+    this.userSpecificDockerRegistryCredentialsProvider =
+        userSpecificDockerRegistryCredentialsProvider;
+    this.clientFactory = clientFactory;
+    this.namespaceFactory = namespaceFactory;
+  }
+
+  @Override
+  public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
+      throws InfrastructureException {
+
+    AuthConfigs authConfigs = userSpecificDockerRegistryCredentialsProvider.getCredentials();
+    if (authConfigs.getConfigs().isEmpty()) {
+      return;
+    }
+
+    Gson gson = new Gson();
```

Comment:
Please explain why do you need to create it here instead of a single line where it is used?